compiler/src/org.graalvm.compiler.asm.aarch64/src/org/graalvm/compiler/asm/aarch64/AArch64Assembler.java

```diff
@@ -1306,7 +1306,9 @@ private void exclusiveLoadInstruction(Instruction instr, Register reg, Register
      */
     private void exclusiveStoreInstruction(Instruction instr, Register rs, Register rt, Register rn, int log2TransferSize) {
         assert log2TransferSize >= 0 && log2TransferSize < 4;
-        assert rt.getRegisterCategory().equals(CPU) && rs.getRegisterCategory().equals(CPU) && !rs.equals(rt);
+        // assert rt.getRegisterCategory().equals(CPU) && rs.getRegisterCategory().equals(CPU) &&
+        // !rs.equals(rt);
```

Comment:
if `!rs.equals(rt)` was indeed unnecessary, we can removed that commented-out line